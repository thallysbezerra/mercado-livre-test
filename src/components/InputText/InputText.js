import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './InputText.scss';

export default class Input extends Component {
	render() {
		const {
			className,
			id,
			onChange,
			onKeyPress,
			placeholder,
			value
		} = this.props;
		return (
			<input
				className={`input-text ${className}`}
				id={id}
				onChange={onChange}
				onKeyPress={onKeyPress}
				placeholder={placeholder}
				type="text"
				value={value}
			/>
		);
	}
}

Input.propTypes = {
	className: PropTypes.string,
	id: PropTypes.string,
	onChange: PropTypes.func,
	onKeyPress: PropTypes.func,
	placeholder: PropTypes.string.isRequired,
	value: PropTypes.string
};

Input.defaultProps = {
	placeholder: 'Needs placeholder text'
};
