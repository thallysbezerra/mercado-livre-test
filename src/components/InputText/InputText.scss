@import '../../css/Theme.scss';

.input-text {
	border: none;
	border-radius: 5px;
	box-sizing: border-box;
	padding: 5px 15px;
	height: 36px;
	transition: box-shadow 0.15s ease-in-out;
	width: 100%;

	&:focus {
		box-shadow: 0 1px 3px 1px rgba(0, 0, 0, 0.1);
		outline: none;
	}

	&::placeholder {
		color: $grey-medium;
	}
}
