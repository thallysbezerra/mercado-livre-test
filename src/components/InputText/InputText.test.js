import React from 'react';
import InputText from './InputText';

import renderer from 'react-test-renderer';

describe('InputText UI', () => {
	it('<InputText /> must have the snapshot version.', () => {
		const tree = renderer.create(<InputText />);

		expect(tree.toJSON()).toMatchSnapshot();
	});
});
