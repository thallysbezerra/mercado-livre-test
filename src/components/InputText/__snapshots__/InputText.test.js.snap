// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputText UI <InputText /> must have the snapshot version. 1`] = `
<input
  className="input-text undefined"
  placeholder="Needs placeholder text"
  type="text"
/>
`;
